lib/mp-readline/readline.c

```diff
@@ -98,6 +99,26 @@ typedef struct _readline_t {
 
 STATIC readline_t rl;
 
+#if MICROPY_REPL_EMACS_WORDS_MOVE
+STATIC size_t cursor_count_word(int forward) {
+    const char *line_buf = vstr_str(rl.line);
+    size_t pos = rl.cursor_pos;
+    bool in_word = false;
+
+    while ((forward || 0 < pos) && (!forward || pos < vstr_len(rl.line))) {
```

Comment:
For me personally, this is just overly succinct and not super readable. Not sure what others think, but both for mainainance and clarity I'd rather have code which speaks for itself instead of having to manually parse 'oh, this actually just means `if(forward) then X else Y`

Reply:
I rewrote this expression more nicely with comments. But it does have to remain somewhat the same to keep the code small..

Reply:
Since you rewrote the expression you don't really need the comments anymore, since the code now is very clear :)